ctapipe/io/eventsource.py

```diff
@@ -210,9 +210,9 @@ def subarray(self) -> SubarrayDescription:
 
     @property
     @abstractmethod
-    def obs_config(self) -> list[ObservationConfigurationContainer]:
+    def observation_config(self) -> Dict[ObservationConfigurationContainer]:
```

Comment:
Isn't the correct type hint `Dict[int, ObservationConfigurationContainer]`? I.e. including the type of the key?

Reply:
Yes: https://docs.python.org/3/library/typing.html#typing.Dict

Reply:
yes, definitely incorrect